chatlearn/synchronizer/parameter_sync.py

```diff
@@ -1172,11 +1241,16 @@ def _multi_thread_sync_for_tp_num_mapping_eq_1(
                     logger.info(f"Sending from {[self.actor2rank[send_actor]]} to {[self.actor2rank[actor] for actor in recv_actors]}.")
                     if self._comm_type == PARAM_SYNC_COMM_TYPE.BROADCAST:
                         actor_groups, finalized_group_name = self.create_broadcast_group(send_actor, recv_actors, param_group=param_group)
+                        if dryrun:
+                            continue
                         futures.append(executor.submit(
-                            self.sync_broadcast, actor_groups, finalized_group_name, requires_grad, filter_fn=filter_fn, param_group=param_group
+                            self.sync_broadcast, actor_groups, finalized_group_name, requires_grad, filter_fn=filter_fn, param_group=param_group,
+                            dryrun=dryrun
```

Comment:
dry_run is always False in this case. Is it the expected behavior?

Reply:
remove the dryrun argument, do not call this function if `dryrun=False`